### PR TITLE
fix issue #436 Color-Bug if ip is an alias

### PR DESCRIPTION
**Pull request by johannes-schmatz**
change all
```
ip link show
```
to something like this
```
(ip -c=never link show || ip link show)
```

### CODE REVIEW

File: bash_completion [hunk 1]
```diff
@@ -917,7 +917,7 @@ _mac_addresses()
     # - ip link: link/ether
     COMPREPLY+=($(
         {
-            LC_ALL=C ifconfig -a || ip link show
+            LC_ALL=C ifconfig -a || (ip -c=never link show || ip link show)
```
Comment (scop) [inline bash_completion:920]: Please remove the parenthesis from everywhere, they're not needed, and actually start an unnecessary subshell.

**Comment by scop** [comment]
completions/ip still needs to be addressed...

**Comment by johannes-schmatz** [comment]
AFAIK completions/ip does call _available_interfaces, so there's no ip invocation

**Comment by scop** [comment]
It calls at least `$1 help`, `$1 r`, and `$1 netns` -- `$1` there is the `ip` command being invoked.

**Comment by johannes-schmatz** [comment]
Thank you! I didn't find them by searching 'ip'

File: completions/ip [hunk 2]
```diff
@@ -48,7 +48,7 @@ _ip()
                 COMPREPLY=($(compgen -W "$c" -- "$cur"))
                 ;;
             *)
-                COMPREPLY=($(compgen -W "help $($1 help 2>&1 | command sed -e \
+                COMPREPLY=($(compgen -W "help $($1 -c=never help 2>&1 || $1 help 2>&1 | command sed -e \
```
Comment (scop) [inline completions/ip:51]: This is not correct; both `$1` calls need to be wrapped in `{ ... }` like `{ $1 -c=never help || $1 help; } 2>&1 || command ...`

File: completions/ip [hunk 3]
```diff
@@ -191,11 +191,11 @@ _ip()
                     ;;
                 a | add | d | del | change | append | r | replace | monitor)
                     if [[ $prev == via ]]; then
-                        COMPREPLY=($(compgen -W "$($1 r | command sed -ne \
+                        COMPREPLY=($(compgen -W "$($1 -c=never r || $1 r | command sed -ne \
```
Comment (scop) [inline completions/ip:194]: Same as above.

File: completions/ip [hunk 4]
```diff
@@ -191,11 +191,11 @@ _ip()
...
                             's/.*via \([0-9.]*\).*/\1/p')" -- "$cur"))
                     elif [[ $prev == "$subcmd" ]]; then
                         COMPREPLY=($(compgen -W "table default \
-                            $($1 r | cut -d ' ' -f 1)" -- "$cur"))
+                            $($1 -c=never r || $1 r | cut -d ' ' -f 1)" -- "$cur"))
```
Comment (scop) [inline completions/ip:198]: Same as above.

File: completions/ip [hunk 5]
```diff
@@ -349,7 +349,7 @@ _ip()
                     ;;
                 delete | exec | pids | set)
                     [[ $prev == "$subcmd" ]] &&
-                        COMPREPLY=($(compgen -W "$($1 netns list)" -- "$cur"))
+                        COMPREPLY=($(compgen -W "$($1 -c=never netns list || $1 netns list)" -- "$cur"))
```
Comment (scop) [inline completions/ip:352]: The first one needs a `2>&1` here.

**Review by scop — CHANGES_REQUESTED** [review]

**Comment by scop** [comment]
Applied with some output redirection fixes to completions/ip.

**Pull request merged**